### PR TITLE
feat(badge): add new component - FE-2627

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
image

### Checklist

- [x] Release notes (Conventional Commits) 
- [x] Unit tests
- [x] Cypress automation tests
- [x] Storybook added or updated
- [x] Theme support
- [x] Typescript `d.ts` file added or updated

### CODE REVIEW

**Comment by DlgSHi** [comment]
added cypress tests

File: cypress/features/regression/test/badge.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
+Feature: Badge component
+  I want to change Badge component properties
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/badge.feature]:
```suggestion
  I want to test Badge component properties
```

File: cypress/features/regression/test/badge.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  Scenario: Badge component rendered properly
+    # commented because of BDD default scenario Given - When - Then
+    # When I open basic Test "Badge" component page
+    Then Badge component rendered properly
+
+  @positive
+  Scenario Outline: Set Badge component to <counter>
+    When I set counter to "<counter>"
+    Then Badge component counter is set to <counter>
+    Examples:
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/badge.feature]: I don't think we need 4 examples. Remove `1` or `5`

File: cypress/features/regression/test/badge.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      | !@#$%^*()_+-=~[];:.,?{} |
+
+  @positive
+  Scenario: Hover Badge component and verify that cross icon appears
+    When I hover onto Badge component
+    Then icon on preview is "cross"
+
+  @positive
+  Scenario: Click event
+    When clear all actions in Actions Tab
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/badge.feature]:
```suggestion
    Given clear all actions in Actions Tab
```

File: cypress/features/regression/test/badge.feature [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      And I click onto Badge component
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/badge.feature]:
```suggestion
      When I click onto Badge component
```

**Comment by harpalsingh** [comment]
A minor adjustment is needed to the badge

Currently:
Screenshot 2020-03-20 at 10 57 59

Update font-size to 12px and the text needs to be 1px up.

Then:
Screenshot 2020-03-20 at 10 56 36

I have tested with 2 digits as well and this is fine.

File: cypress/features/regression/test/badge.feature (hunk 1 above)
Comment (DlgSHi) [reply cypress/features/regression/test/badge.feature]: updated

File: cypress/features/regression/test/badge.feature (hunk 2 above)
Comment (DlgSHi) [reply cypress/features/regression/test/badge.feature]: updated

File: cypress/features/regression/test/badge.feature (hunk 3 above)
Comment (DlgSHi) [reply cypress/features/regression/test/badge.feature]: updated

File: cypress/features/regression/test/badge.feature (hunk 4 above)
Comment (DlgSHi) [reply cypress/features/regression/test/badge.feature]: updated

File: cypress/features/regression/test/badge.feature [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    Examples:
+      | counter |
+      | 101     |
+      | 250     |
+      | 999     |
+
+  @negative
+  Scenario Outline: Set Badge component to out of scope to <counter>
+    When I set counter to "<counter>"
+    Then Badge component counter is not exist
```
Comment (damianOfKyan) [inline cypress/features/regression/test/badge.feature]:
```suggestion
    Then Badge component counter does not exist
```

Comment (DlgSHi) [reply cypress/features/regression/test/badge.feature]: updated, thanks

**Comment by harpalsingh** [comment]
had a another view after code changes, all good 👍

**Pull request merged**